File: visa-dev-samplecode-master/vdp-java/src/test/java/com/visa/vdp/visacheckout/TestUpdatePaymentInformation.java
package com.visa.vdp.visacheckout;

import java.util.HashMap;

import org.apache.commons.lang3.StringUtils;
import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;
import org.apache.http.HttpStatus;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

import com.visa.vdp.utils.VisaAPIClient;
import com.visa.vdp.utils.MethodTypes;
import com.visa.vdp.utils.Property;
import com.visa.vdp.utils.VisaProperties;

public class TestUpdatePaymentInformation {

    String apiKey;
    String updatePaymentInfoRequest;
    VisaAPIClient visaAPIClient;

    @BeforeTest(groups = "visacheckout")
    public void setup() {
        this.visaAPIClient = new VisaAPIClient();
        this.apiKey = VisaProperties.getProperty(Property.API_KEY);
        this.updatePaymentInfoRequest = "{"
		                 + "\"orderInfo\": {"
		                 + "\"currencyCode\": \"USD\","
		                 + "\"discount\": \"5.25\","
		                 + "\"eventType\": \"Confirm\","
		                 + "\"giftWrap\": \"10.1\","
		                 + "\"misc\": \"3.2\","
		                 + "\"orderId\": \"testorderID\","
		                 + "\"promoCode\": \"testPromoCode\","
		                 + "\"reason\": \"Order Successfully Created\","
		                 + "\"shippingHandling\": \"5.1\","
		                 + "\"subtotal\": \"80.1\","
		                 + "\"tax\": \"7.1\","
		                 + "\"total\": \"101\""
		               + "}"
		            + "}";
    }

    @Test(groups = "visacheckout")
    public void testUpdatePaymentInfo() throws Exception {
        String baseUri = "wallet-services-web/";
        String resourcePath = "payment/info/{callId}";
        resourcePath = StringUtils.replace(resourcePath, "{callId}", VisaProperties.getProperty(Property.CHECKOUT_CALL_ID));

        CloseableHttpResponse response = this.visaAPIClient.doXPayTokenRequest(baseUri, resourcePath, "apikey=" + this.apiKey, "Update Payment Information Test", this.updatePaymentInfoRequest, MethodTypes.PUT, new HashMap<String, String>());
        Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
        response.close();
    }
}

File: visa-dev-samplecode-master/vdp-python/visa/test/visacheckout/test_update_payment_information.py
from visa.helpers.visa_api_client import VisaAPIClient
import json
import sys
import os
import unittest
if sys.version_info < (3, 0):
    import ConfigParser as parser
else:
    import configparser as parser
'''
@author: visa
'''

class TestUpdatePaymentInformation(unittest.TestCase):
    
    config = parser.ConfigParser()
    config_path = os.path.abspath(os.path.join(os.path.dirname(os.path.dirname(__file__)),'..','configuration.ini'))
    config.read(config_path)
    
    def setUp(self):
        self.visa_api_client = VisaAPIClient()
        self.update_payment_info_request = json.loads('''{
                          "orderInfo": {
                          "currencyCode": "USD",
                          "discount": "5.25",
                          "eventType": "Confirm",
                          "giftWrap": "10.1",
                          "misc": "3.2",
                          "orderId": "testorderID",
                          "promoCode": "testPromoCode",
                          "reason": "Order Successfully Created",
                          "shippingHandling": "5.1",
                          "subtotal": "80.1",
                          "tax": "7.1",
                          "total": "101"
                        }
                     }''')

    def test_update_payment_info(self):
        base_uri = 'wallet-services-web/'
        resource_path = 'payment/info/{callId}'
        resource_path = resource_path.replace('{callId}', self.config.get('VDP','checkoutCallId'))
        query_string = 'apikey=' + self.config.get('VDP','apiKey')
        response = self.visa_api_client.do_x_pay_request(base_uri, resource_path , query_string, self.update_payment_info_request, 'Update Payment Information Test', 'put')
        self.assertEqual(str(response.status_code) ,"200" ,"Update Payment Information test failed")
        pass
